lib/module.ts

```diff
@@ -13,20 +13,20 @@ export interface IModule<
   S extends IState,
   G extends IGetters,
   M extends IMutations,
-  SM extends ISubModules,
+  SM extends ISubModules = never,
 > {
   readonly options: IModuleOptions<S, G, M, SM>
   flatten(): IFlattenedModule<S, G, M, SM>
 }
 
-/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-empty-interface */
-export interface ISubModules extends Record<string, IModule<any, any, any, any>> {}
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ISubModules = Record<string, IModule<any, any, any, any>>
```

Comment:
I'm less sure about using `type` instead of `interface` here, but I think it should be ok. The only restriction I'm aware of is that types don't allow [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html).

Reply:
This declaration merging seems strange, more like "I like to give all stuff the same name" to me.
But I know it's here because developers do need it at some point, and I think I've seen something like this in some source codes a few times.
Just curious, have you ever found yourself in a situation that you must use this feature to achieve what you want?

Reply:
I have not! Anyway, it's easy enough to refactor back to `interface` later if needed.

Reply:
Got it, thanks!